cli/src/commands/content-directory/createEntity.ts

```diff
@@ -0,0 +1,170 @@
+            return answer === 'true' || false
+          },
+        }),
+      }
+
+      const isQuestionOptional = propertySubtype === 'Bool' ? '' : required.isTrue ? '(required)' : '(optional)'
+      const classId = isSubtypeReference
+        ? ` [Class Id: ${propertyType.asType('Single').asType('Reference')[0].toString()}]`
+        : ''
+
+      return {
+        name: name.toString(),
+        message: `${name} - ${propertySubtype}${classId} ${isQuestionOptional}`,
+        type: questionType,
+        ...optionalQuestionProperties,
+      }
+    })
+  }
+
+  checkReferencesValidity = async (
+    answers: {
+      [key: string]: string | number | null
+    },
+    { properties }: ContentDirectoryClass
+  ) => {
+    const propertyReferenceClassIds: (string | null)[] = []
+
+    const references = await Promise.all(
+      Object.keys(answers).map((key: string, index) => {
+        const propertyType = properties[index].property_type
+
+        if (propertyType.isOfType('Single') && propertyType.asType('Single').isOfType('Reference')) {
+          if (answers[key] !== null) {
+            propertyReferenceClassIds.push(propertyType.asType('Single').asType('Reference')[0].toString())
+            return this.getEntity(answers[key] as number | string)
+          }
+        }
+
+        propertyReferenceClassIds.push(null)
+      })
+    )
+
+    references.forEach((reference, index) => {
+      if (reference) {
+        if (reference.class_id.toString() !== propertyReferenceClassIds[index]) {
+          throw new CLIError(`The #${propertyReferenceClassIds[index]} entity is not of the right class id!`)
+        }
+      }
+    })
+  }
+
+  async getActor(context: typeof CONTEXTS[number], pickedClass: ContentDirectoryClass) {
+    if (context === 'Member') {
+      if (pickedClass.class_permissions.any_member.isFalse) {
+        this.error(`You're not allowed to createEntity of className: ${pickedClass.name.toString()}!`)
+      }
+
+      const memberId = await this.getRequiredMemberId()
+
+      return this.createType('Actor', { Member: memberId })
+    } else if (context === 'Curator') {
+      return await this.getCuratorContext([pickedClass.name.toString()])
+    } else {
+      await this.getRequiredLead()
+
+      return this.createType('Actor', { Lead: null })
+    }
+  }
+
+  async run() {
+    const { className } = this.parse(CreateEntityCommand).args as CreateEntityArgs
+    const { context } = this.parse(CreateEntityCommand).flags as CreateEntityFlags
+
+    const currentAccount = await this.getRequiredSelectedAccount()
+    await this.requestAccountDecoding(currentAccount)
+    const [, entityClass] = await this.classEntryByNameOrId(className)
+
+    const actor = await this.getActor(context, entityClass)
+
+    const answers: {
+      [key: string]: string | number | null
+    } = await inquirer.prompt(this.getQuestionsFromClass(entityClass))
+
+    await this.checkReferencesValidity(answers, entityClass)
```

Comment:
I think if we use inquirer question `validate` property for this we can validate it more "dynamically" instead of waiting until all questions are answered (and in that case the user wouldn't need to provide the entire input again in case of an error, which seems like a nice improvement)